src/library/export/trackexportdlg.cpp

```diff
@@ -48,6 +52,47 @@ void TrackExportDlg::showEvent(QShowEvent* event) {
         qDebug() << "Programming error: did not initialize m_exporter, about to crash";
         return;
     }
+
+    //Checks if there are any files that are missing. If so, it will throw dialog informing the user.
+    //User can then decide to either cancel the export or skip those files
+    QList<QString> files = m_worker->getMissingTracks();
+    {
+        if (!files.isEmpty()) {
+            QDialog notExportedDlg = QDialog();
+            notExportedDlg.setWindowTitle(tr("Export Track Files"));
+            QVBoxLayout notExportedLayout;
+            QLabel notExportedLabel;
+            notExportedLabel.setText(
+                    tr("The following %1 files were not found at the specified "
+                       "file location and as a result were not exported."
+                       "Click \"OK\" to skip these files. Click \"Cancel\" to cancel the export.")
+                            .arg(QString::number(files.length())));
+            notExportedLabel.setTextFormat(Qt::RichText);
+
+            QListWidget notExportedList;
+            notExportedList.addItems(files);
+
+            QDialogButtonBox exportedDlgButtons = QDialogButtonBox();
+            exportedDlgButtons.addButton(QDialogButtonBox::Ok);
+            exportedDlgButtons.addButton(QDialogButtonBox::Cancel);
+            connect(&exportedDlgButtons,
+                    &QDialogButtonBox::accepted,
+                    &notExportedDlg,
+                    &QDialog::accept);
+            connect(&exportedDlgButtons,
+                    &QDialogButtonBox::rejected,
+                    &notExportedDlg,
+                    &QDialog::reject);
+
+            notExportedLayout.addWidget(&notExportedLabel);
+            notExportedLayout.addWidget(&notExportedList);
+            notExportedLayout.addWidget(&exportedDlgButtons);
+            notExportedDlg.setLayout(&notExportedLayout);
+            if (notExportedDlg.exec() == 0) {
```

Comment:
Don't use magic numbers

https://doc.qt.io/qt-5/qdialog.html#DialogCode-enum

Reply:
so is it better to have 
```
notExportedDlg.exec()
if(notExportedDlg.rejected()) {
```

Reply:
Please read the documentation carefully: https://doc.qt.io/qt-5/qdialog.html#exec

> The function returns a DialogCode result.

Reply:
Ah I see what you mean here